compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1025,7 +1025,13 @@ fn adt_def(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::AdtDef<'_> {
 
     let is_anonymous = item.ident.name == kw::Empty;
     let repr = if is_anonymous {
-        tcx.adt_def(tcx.local_parent(def_id)).repr()
+        let parent = tcx.local_parent(def_id);
+        if matches!(tcx.hir_node_by_def_id(parent), Node::Item(_)) {
```

Comment:
I would probably make this a bit stricter:

```suggestion
        if let Node::Item(item) = tcx.hir_node_by_def_id(parent) && item.is_adt() {
```

or better yet

```suggestion
        if let Node::Item(item) = tcx.hir_node_by_def_id(parent)
            && let ItemKind::Struct(..) | ItemKind::Union(..) = item.kind
        {
```

Reply:
However, I'm wondering if it's always the case that the parent is a `Node::Item`? I mean CI does pass so it seems to be fine. Since I've only skimmed the parsing & lowering PR for `unnamed_fields`, I presume that the item kind of anon adts themselves is `ItemKind::Struct | ItemKind::Union`? This is relevant for

```rs
#[repr(C) struct S { _: struct { _: struct { x: i32 } } }
```

Reply:
That's how I understand it, yeah